### PR TITLE
Feature - My Lists

### PR DESCRIPTION
**Pull request by kosovych**
- fixed pagination bug
- added 'My Lists'

### CODE REVIEW

File: src/store/userLists/operations/createUserListOperation.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        }
+      })
+      cb()
+    } catch (error) {
+      openNotification({
+        message: error.message,
+        type: 'error'
+      })
+    }
+    dispatch(createUserListFinish())
```
Comment (VladWithCookies) [inline src/store/userLists/operations/createUserListOperation.js:35]: are you sure this action need to be called outside `try` block?

File: src/containers/UserLists/component.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  <>
+    <Row>
+      <Col
+        offset={2}
+        span={20}
+      >
+        <div className="top-margin">
+          <Typography.Title>
+            My Lists
+            {' '}
```
[the remark this answers is not in the document]
Comment (kosovych) [reply src/containers/UserLists/component.js:30]: @VladWithCookies this is from stubs. In my opinion no need to waste time on the layout

File: src/store/userLists/actions.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+export const requestUserListsError = () => ({
+  type: REQUEST_USER_LISTS_ERROR
+})
+
+export const getUserLists = page => ({
+  type: GET_USER_LISTS,
+  page
+})
+
+export const createUserLits = (values, cb) => ({
```
[the remark this answers is not in the document]
Comment (kosovych) [reply src/store/userLists/actions.js:30]: @VladWithCookies agree with you, but I think everybody knows `cb` it's `callback`

File: src/containers/Dashboard/component.js [hunk 4]
```diff
@@ -59,16 +59,17 @@ const Dashboard = ({
         justify="center"
       >
         <Col>
-          {movies && !!totalPages && (
+          {movies.length ? (
```
[the remark this answers is not in the document]
Comment (kosovych) [reply src/containers/Dashboard/component.js:62]: ✅

File: src/containers/Favorites/component.js [hunk 5]
```diff
@@ -61,16 +61,17 @@ const WatchList = ({
       justify="center"
     >
       <Col>
-        {movies && !!totalPages && (
+        {movies.length ? (
```
[the remark this answers is not in the document]
Comment (kosovych) [reply src/containers/Favorites/component.js:64]: ✅

File: src/containers/UserLists/CreateUserListModal/container.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+import * as Yup from 'yup'
+import { connect } from 'react-redux'
+
+import { createListLoadingSelector } from 'Store/userLists/selectors'
+import { createUserLits as createUserLitsAction } from 'Store/userLists/actions'
+import CreateUserListModalComponent from './component'
+
+const UserListSchema = Yup.object().shape({
+  name: Yup.string()
+    .matches(/^[A-Za-z0-9\s]+$/, { message: 'Should be combination of numbers & alphabets' })
```
[the remark this answers is not in the document]
Comment (kosovych) [reply src/containers/UserLists/CreateUserListModal/container.js:13]: ✅

File: src/containers/UserLists/CreateUserListModal/container.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+  createUserLits: createUserLitsAction
+}
+
+export default connect(mapStateToProps, mapDispatchToProps)(withFormik({
+  mapPropsToValues: () => ({ name: '', description: '' }),
+  handleSubmit: (values, { props, resetForm }) => {
+    const { createUserLits, toggleCreateListModalOpen } = props
+    createUserLits(values, () => {
+      resetForm()
+      toggleCreateListModalOpen()
```
[the remark this answers is not in the document]
Comment (kosovych) [reply src/containers/UserLists/CreateUserListModal/container.js:59]: @VladWithCookies Yes, we can call those functions inside form handler functions, but in error case user can retry to send the form immediately without reentering text and reopening the modal window.

File: src/containers/UserLists/List/DeleteUserList/component.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+const DeleteUserList = ({ onDeleteList }) => (
+  <button
+    type="button"
+    aria-label="Remove List"
+    className="reset-btn"
+    onClick={onDeleteList}
+  >
+    <DeleteOutlined
+      key="delete"
```
[the remark this answers is not in the document]
Comment (kosovych) [reply src/containers/UserLists/List/DeleteUserList/component.js:13]: no

File: src/containers/UserLists/List/DeleteUserList/container.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+import { removeUserList as removeUserListAction } from 'Store/userLists/actions'
+import DeleteUserListComponent from './component'
+
+class DeleteUserList extends Component {
+  onDeleteList = () => {
+    const { listId, removeUserList } = this.props
+    Modal.confirm({
+      title: 'Do you want to delete list?',
+      onOk() { removeUserList(listId) },
+      onCancel() {}
```
[the remark this answers is not in the document]
Comment (kosovych) [reply src/containers/UserLists/List/DeleteUserList/container.js:15]: yes

File: src/containers/UserLists/List/component.js [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+      lg={{ span: 8 }}
+      xl={{ span: 6 }}
+    >
+      <Card
+        hoverable
+        className="top-margin"
+        actions={[
+          <DeleteUserList
+            listId={id}
+            key="delete"
```
[the remark this answers is not in the document]
Comment (kosovych) [reply src/containers/UserLists/List/component.js:24]: no, we can't. Every React element in the array should have the `key`

File: src/containers/UserLists/component.js [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+          )}
+        </Row>
+      </Col>
+    </Row>
+    <Row
+      type="flex"
+      justify="center"
+    >
+      <Col>
+        {userLists.length ? (
```
[the remark this answers is not in the document]
Comment (kosovych) [reply src/containers/UserLists/component.js:88]: good point

File: src/containers/UserLists/container.js [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+
+  componentDidMount() {
+    const { getUserLists } = this.props
+    getUserLists()
+  }
+
+  onPageChange = (page) => {
+    const { getUserLists } = this.props
+    getUserLists(page)
+  }
```
[the remark this answers is not in the document]
Comment (kosovych) [reply src/containers/UserLists/container.js:28]: ✅

File: src/containers/UserLists/container.js [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+
+  toggleCreateListModalOpen = () => {
```
[the remark this answers is not in the document]
Comment (kosovych) [reply src/containers/UserLists/container.js:30]: ✅

File: src/store/rootReducer.js [hunk 14]
```diff
@@ -5,13 +5,15 @@ import data from './data'
...
+import userLists from './userLists'
 
 const reducers = combineReducers({
   auth,
   data,
   trendingMovies,
   watchList,
-  favorites
+  favorites,
+  userLists
```
[the remark this answers is not in the document]
Comment (kosovych) [reply src/store/rootReducer.js:16]: ✅

File: src/store/userLists/operations/createUserListOperation.js (hunk 1 above)
Comment (kosovych) [reply src/store/userLists/operations/createUserListOperation.js:35]: yes, I need to set `false` for loading state

File: src/containers/UserLists/container.js [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+  state = {
+    createListModalOpen: false
+  }
+
+  componentDidMount() {
+    const { getUserLists } = this.props
+    getUserLists()
+  }
+
+  handleYourFunctionName = () => {
```
Comment (VladWithCookies) [inline src/containers/UserLists/container.js:25]: `handleYourFunctionName` is just an example

File: src/containers/UserLists/CreateUserListModal/container.js [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+import PropTypes from 'prop-types'
+import { withFormik } from 'formik'
+import * as Yup from 'yup'
+import { connect } from 'react-redux'
+
+import { createListLoadingSelector } from 'Store/userMovieLists/selectors'
+import { createUserLits as createUserLitsAction } from 'Store/userMovieLists/actions'
+import CreateUserListModalComponent from './component'
+
+const regexp = /^[A-Za-z0-9\s]+$/
```
Comment (VladWithCookies) [inline src/containers/UserLists/CreateUserListModal/container.js:11]: extract to constants file

File: src/containers/UserLists/component.js [hunk 17]
```diff
@@ -0,0 +1,134 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import {
+  Row, Col, Typography, Pagination, Empty, Spin, Alert
+} from 'antd'
+import { PlusCircleOutlined } from '@ant-design/icons'
+import List from './List'
+import CreateUserListModal from './CreateUserListModal'
+
+
```
Comment (VladWithCookies) [inline src/containers/UserLists/component.js:10]: extra new line

File: src/containers/WatchList/component.js [hunk 18]
```diff
@@ -61,16 +61,17 @@ const WatchList = ({
       justify="center"
     >
       <Col>
-        {movies && !!totalPages && (
+        {movies.length ? (
```
Comment (VladWithCookies) [inline src/containers/WatchList/component.js:64]: `Boolean(movies.length) && ( ...`

File: src/store/logics.js [hunk 19]
```diff
@@ -2,6 +2,11 @@ import { loginOperation, getUserOperation } from './auth/operations'
 import { getMoviesOperation } from './trendingMovies/operations'
 import { getWatchListOperation, removeMovieOperation } from './watchList/operations'
 import { getFavoritesOperation, removeFavoriteMovieOperation } from './favorites/operations'
+import {
+  createUserListOperation,
+  getUserListsOperations,
```
Comment (VladWithCookies) [inline src/store/logics.js:7]: `getUserListsOperation`

File: src/store/userLists/operations/createUserListOperation.js (hunk 1 above)
Comment (VladWithCookies) [reply src/store/userLists/operations/createUserListOperation.js]: need to be discussed

File: src/containers/UserLists/CreateUserListModal/container.js (hunk 16 above)
Comment (kosovych) [reply src/containers/UserLists/CreateUserListModal/container.js:11]: @VladWithCookies this variable are used only for this component, should I move it to the constants file anyway?

File: src/containers/UserLists/component.js (hunk 17 above)
Comment (kosovych) [reply src/containers/UserLists/component.js:10]: ✅

File: src/containers/WatchList/component.js (hunk 18 above)
Comment (kosovych) [reply src/containers/WatchList/component.js:64]: ✅

File: src/store/logics.js (hunk 19 above)
Comment (kosovych) [reply src/store/logics.js:7]: ✅

File: src/containers/UserLists/CreateUserListModal/container.js (hunk 16 above)
Comment (kosovych) [reply src/containers/UserLists/CreateUserListModal/container.js:11]: ✅

File: src/containers/UserLists/container.js (hunk 15 above)
Comment (kosovych) [reply src/containers/UserLists/container.js:25]: ✅

File: src/store/userLists/operations/createUserListOperation.js (hunk 1 above)
Comment (kosovych) [reply src/store/userLists/operations/createUserListOperation.js]: ✅

**Review by VladWithCookies — APPROVED** [review]

**Pull request merged**